client/mainmenu/CHighScoreScreen.cpp

```diff
@@ -289,6 +291,9 @@ int CHighScoreInputScreen::addEntry(std::string text) {
 
 void CHighScoreInputScreen::show(Canvas & to)
 {
+	if(background)
+		background->redraw();
```

Comment:
Please don't use 'redraw' method  inside show/showAll calls.
showAll(to) is more suitable when simple show(to) is not enough

(same goes to other such cases)